backend/electrum_backend.go

```diff
@@ -257,29 +265,38 @@ func (eb *ElectrumBackend) processTxRequest(node *electrum.Node, txHash string)
 		eb.removeNode(node.Ident)
 
 		// requeue request
-		// TODO: we should have a retry counter and fail gracefully if an address fails too
-		// many times.
+		// TODO: we should have a retry counter and fail gracefully if a transaction fails
+		//       too many times.
 		eb.txRequests <- txHash
 		return err
 	}
+	height, err := eb.getTxHeight(txHash)
+	if err != nil {
+		// a transaction thas is being asked for is not in the cache, yet (most likely)
```

Comment:
Have you figured out why this could happen? For now we can have a panic() here if the cache is inconsistent and I can debug later.